Sources/Penny/Handlers/InteractionHandler.swift

```diff
@@ -850,7 +1123,36 @@ private extension InteractionHandler {
             }
         )
     }
-    
+
+    func handleAutoFaqsAutocomplete(
+        data: Interaction.ApplicationCommand
+    ) async throws -> Payloads.InteractionResponse.Autocomplete {
+        let first = try (data.options?.first).requireValue()
+        let expression = try first.options
+            .requireValue()
+            .requireOption(named: "expression")
+            .requireString()
+        let foldedExpression = expression.heavyFolded()
+        let all = try await context.services.autoFaqsService.getAll().map(\.key)
+        let queried: ArraySlice<String>
+        if foldedExpression.isEmpty {
+            queried = ArraySlice(all.sorted { $0 > $1 }.prefix(25))
+        } else {
+            queried = all
+                .filter { $0.heavyFolded().contains(foldedExpression) }
+                .sorted { $0 > $1 }
+                .prefix(25)
+        }
```

Comment:
```suggestion
        let queried =
                (!foldedExpression.isEmpty ? all.heavyFolded().filter { $0.contains(foldedExpression) } : all)
                .sorted { $0 > $1 }
                .prefix(25)
```

Reply:
I don't like the original code, but i don't think i like this new code either 😅

Reply:
I think this is a Set, not a String, so heavy folded is called for each string not each character 😅 the array slice probably threw you off